atomic_reactor/core.py

```diff
@@ -272,10 +277,60 @@ def __init__(self, base_url=None, retry_times=DOCKER_MAX_RETRIES,
 
         if hasattr(docker, 'AutoVersionClient'):
             client_kwargs['version'] = 'auto'
-        client_kwargs['retry'] = retry_times
+        self.retry_times = retry_times
+        client_kwargs['retry'] = self.retry_times
 
         self.d = WrappedDocker(**client_kwargs)
 
+    def retry_generator(self, fnc_name, function, *args, **kwargs):
```

Comment:
Since `fnc_name` is not human-readable anyway, lets use `function.__name__`, since all the functions are from `self.d`

Reply:
I can't use __name__ as that will just show 'hooked some reference'

Reply:
I'm pretty sure it works:
```
>>> import docker
>>> client = docker.APIClient()
>>> client.pull.__name__
'pull'
```

Reply:
yeah but self.d is already wrapped, I've tried using __ name __ and didn't work

Reply:
That's because `WrappedDocker.__getattr__`'s `hooked` doesn't preseve `orig_attr`'s `__name__` - it seems to be a one-line change to me

Reply:
ok will preserve it